includes/woopay/class-woopay-session.php

```diff
@@ -583,4 +583,24 @@ private static function create_woopay_nonce( int $uid ) {
 		return substr( wp_hash( $i . '|' . $action . '|' . $uid . '|' . $token, 'nonce' ), -12, 10 );
 	}
 
+	/**
+	 * Gets the custom message from the settings and replaces the placeholders with the correct links.
+	 *
+	 * @return string The custom message with the placeholders replaced.
+	 */
+	private static function get_formatted_custom_message() {
+		$custom_message = WC_Payments::get_gateway()->get_option( 'platform_checkout_custom_message' );
+
+		$replacement_map = [
+			'[terms_of_service_link]' => wc_terms_and_conditions_page_id() ?
+				'<a href="' . get_permalink( wc_terms_and_conditions_page_id() ) . '">' . __( 'Terms of Service', 'woocommerce-payments' ) . '</a>' :
+				__( 'Terms of Service', 'woocommerce-payments' ),
+			'[privacy_policy_link]'   => wc_privacy_policy_page_id() ?
+				'<a href="' . get_permalink( wc_privacy_policy_page_id() ) . '">' . __( 'Privacy Policy', 'woocommerce-payments' ) . '</a>' :
+				__( 'Privacy Policy', 'woocommerce-payments' ),
```

Comment:
```suggestion
				'<a href="' . get_permalink( wc_terms_and_conditions_page_id() ) . '" target="_blank" rel="noopener noreferrer">' . __( 'Terms of Service', 'woocommerce-payments' ) . '</a>' :
				__( 'Terms of Service', 'woocommerce-payments' ),
			'[privacy_policy_link]'   => wc_privacy_policy_page_id() ?
				'<a href="' . get_permalink( wc_privacy_policy_page_id() ) . '" target="_blank" rel="noopener noreferrer">' . __( 'Privacy Policy', 'woocommerce-payments' ) . '</a>' :
				__( 'Privacy Policy', 'woocommerce-payments' ),
```

Reply:
I was just wondering if it might be better to have these links open in a new tab and then I rechecked the [acceptance criteria](https://github.com/Automattic/woocommerce-payments/issues/6993#issue-1850088254) and it looks like that was the original intention as well.

Reply:
I missed that 😓, thanks!